packages/core/metrics/core-metrics-server/src/metrics.ts

```diff
@@ -40,6 +40,46 @@ export interface IntervalHistogram {
   };
 }
 
+/**
+ * Metrics related to the elasticsearch clients
+ * @public
+ */
+export interface ElasticsearchClientsMetrics {
+  /** Number of HTTP Agents that have are currently being used by the ES-js client */
+  agents: number;
+  /** Number of ES instances (or proxies) that ES-js client is connecting to */
+  connectedNodes: number;
+  /** Number of nodes with active connections */
+  nodesWithActiveSockets: number;
+  /** Number of nodes with available connections (alive but idle).
+   * Note that a node can have both active and idle connections at the same time
+   */
+  nodesWithIdleSockets: number;
+  /** Total number of active sockets (all nodes, all connections) */
+  totalActiveSockets: number;
+  /** Total number of available sockets (alive but idle, all nodes, all connections) */
+  totalIdleSockets: number;
+  /** Total number of queued requests (all nodes, all connections) */
+  totalQueuedRequests: number;
+  /** Number of active connections of the node with most active connections */
+  mostActiveNodeSockets: number;
+  /** Average of active sockets per node (all connections) */
+  averageActiveSocketsPerNode: number;
+  /** Number of idle connections of the node with most idle connections */
+  mostIdleNodeSockets: number;
+  /** Average of available (idle) sockets per node (all connections) */
+  averageIdleSocketsPerNode: number;
+}
+
+/**
+ * Metrics related to the elasticsearch clients, by protocol
+ * @public
+ */
+export interface ElasticsearchClientsMetricsByProtocol {
+  http: ElasticsearchClientsMetrics;
+  https: ElasticsearchClientsMetrics;
+}
```

Comment:
What's the value in having separate metrics for http and https?
Is it even possible to configure Kibana to use https and http nodes? Is it even possible to run an Elasticsearch cluster with mixed http / https nodes?

As a first step we just want to use this data to understand and optimize the default performance on cloud. But if this is useful to us it's useful to customers too. But if there's two keys then all the dashboards built on this data will have to have two queries and it will always show e.g. "http.averageActiveSocketsPerNode" and "https.averageActiveSocketsPerNode" even if only one will have data. So this kinda makes the dashboard noisy.

Reply:
I think it's technically possible, at least from a configuration standpoint. When users specify the `hosts: ` property they could be introducing mixed http and https hosts.

I was thinking that we would mainly rely on "https" for dashboards, which will be the normal for our cloud deployments.

Then, if we find a faulty deployment with degraded performance, we could take a look and see if there are open sockets in the 'http' protocol? I'm not really sure how helpful it can be.

If you think it brings more noise than value I can simply remove the 'http' one, or try to merge them together into a single "store". Alternatively, from the Kibana metrics beat, we could simply consume the 'https' one. This way, the `/api/stats` will continue to expose both (which could be useful for investigations on self-managed), and the monitoring cluster will only ingest the 'https' one, WDYT?

Reply:
I can imagine it might be possible to configure mixed http/https nodes for Elasticsearch for high availability you might have mixed nodes while you switch nodes in your cluster one at a time to serving over https. But I suspect this is extremely rare as a long term configuration.

Mixed protocols means the `elasticsearch.maxSockets` configuration option behaves differently and Kibana would have double as many sockets as a pure http or pure https configuration. In such a case only seeing the https sockets in a dashboard would be misleading and I think it would be more useful to see all sockets to all protocols on a dashboard.

So it feels like merging the agents would always provide a more holistic picture of how many sockets your Kibana is opening. I'm not sure if knowing if it's https or http would make us make any different decisions, in some sense it feels like an implementation detail, but we could have a field like `protocol: 'http' | 'https' | 'mixed'`.

Reply:
That's a fair point, I'll update my PR accordingly, thanks!